packages/api/src/clients/client-settings.ts

```diff
@@ -0,0 +1,13 @@
+export type ClientSettings = {
+  /**
+   * the URL to use for managing user oauth tokens.
+   * Specify this value if you are using a regional bot.
+   * For e.g., https://europe.token.botframework.com
+   * Default is https://token.botframework.com
+   */
+  readonly OAuthUrl?: string;
+};
+
+export const DEFAULT_CLIENT_SETTINGS: Required<ClientSettings> = {
```

Comment:
NIT: Instead of making all keys optional in ClientSettings, and making DEFAULT_CLIENT_SETTINGS `Required`, you could make all keys required in ClientSettings, and make all parameters `Partial`. This way, in places you need `ClientSettings`, the value of the keys isn't string | undefined.
For eg, in sign-in.ts, you have:
```ts
`${this._clientSettings.OAuthUrl}/${BOT_SIGNIN_ENDPOINTS.URL}?${q}`
```
here OauthUrl is actually string | undefined, which is less understandable than if it was just string.